/*
 * This file was automatically generated by EvoSuite
 * Fri Nov 26 07:48:11 GMT 2021
 */

package com.gildedrose;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.gildedrose.ProductListException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ProductListException_ESTest extends ProductListException_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      ProductListException productListException0 = null;
      try {
        productListException0 = new ProductListException("", (Throwable) null, false, false);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0
         //
         verifyException("org.evosuite.runtime.mock.java.lang.MockException", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ProductListException productListException0 = new ProductListException();
      ProductListException productListException1 = new ProductListException("0<Cf{0A", productListException0, true, true);
      assertFalse(productListException1.equals((Object)productListException0));
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ProductListException productListException0 = new ProductListException("0<Cf{0A");
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ProductListException productListException0 = new ProductListException();
      ProductListException productListException1 = new ProductListException("0<Cf{0A", productListException0);
      assertFalse(productListException1.equals((Object)productListException0));
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      ProductListException productListException0 = new ProductListException();
      ProductListException productListException1 = new ProductListException(productListException0);
      assertFalse(productListException1.equals((Object)productListException0));
  }
}
